pinot-core/src/main/java/org/apache/pinot/core/query/selection/SelectionOperatorService.java

```diff
@@ -78,6 +77,8 @@ public SelectionOperatorService(QueryContext queryContext, DataSchema dataSchema
     _offset = queryContext.getOffset();
     _numRowsToKeep = _offset + queryContext.getLimit();
     assert queryContext.getOrderByExpressions() != null;
+    // TODO: Do not use type compatible comparator for performance since we don't support different data schema on
```

Comment:
qq: can't we add a flag in the dataTable metadata to indicate whether a block is sorted?

Reply:
Good point. Added it in the `TODO`